docs/README.md

```diff
@@ -0,0 +1,320 @@
+        name = name,
+        args = args,
+        # Always run tsc with the working directory in the project folder
+        chdir = native.package_name(),
+        **kwargs
+    )
+```
+
+so that the users `BUILD` file can omit some of the syntax and default settings:
+
+```starlark
+load(":tsc.bzl", "tsc")
+
+tsc(
+    name = "two",
+    srcs = [
+        "tsconfig.json",
+        "two.ts",
+        "//:node_modules/@types/node",
+        "//examples/js_library/one",
+    ],
+    outs = [
+        "two.js",
+    ],
+)
+```
+
+### Custom rules
+
+If macros are not sufficient to express your Bazel logic, you can use a custom rule instead.
+Aspect has written a number of these based on rules_js, such as:
+
+-   [rules_ts](https://github.com/aspect-build/rules_ts) - Bazel rules for the `tsc` compiler from <http://typescriptlang.org>
+-   [rules_swc](https://github.com/aspect-build/rules_swc) - Bazel rules for the swc toolchain <https://swc.rs/>
+-   [rules_jest](https://github.com/aspect-build/rules_jest) - Bazel rules to run tests using https://jestjs.io
+-   [rules_esbuild](https://github.com/aspect-build/rules_esbuild) - Bazel rules for <https://esbuild.github.io/> JS bundler
+-   [rules_webpack](https://github.com/aspect-build/rules_webpack) - Bazel rules for webpack bundler <https://webpack.js.org/>
+-   [rules_terser](https://github.com/aspect-build/rules_terser) - Bazel rules for <https://terser.org/> - a JavaScript minifier
+-   [rules_rollup](https://github.com/aspect-build/rules_rollup) - Bazel rules for <https://rollupjs.org/> - a JavaScript bundler
+-   [rules_deno](https://github.com/aspect-build/rules_deno) - Bazel rules for Deno http://deno.land
+
+You can also write your own custom rule, though this is an advanced topic and not covered in this documentation.
+
+### Documenting your macros and custom rules
+
+You can use [stardoc] to produce API documentation from Starlark code.
+We recommend producing Markdown output, and checking those `.md` files into your source repository.
+This makes it easy to browse them at the same revision as the sources.
+
+You'll need to create `bzl_library` targets for your starlark files.
+This is a good practice as it lets users of your code generate their own documentation as well.
+
+In addition, Aspect's bazel-lib provides some helpers that make it easy to run stardoc and check that it's up-to-date.
+
+Continuing our example, where we wrote a macro in `tsc.bzl`, we'd write this to document it, in `BUILD`:
+
+```starlark
+load("@aspect_bazel_lib//lib:docs.bzl", "stardoc_with_diff_test", "update_docs")
+load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+
+bzl_library(
+    name = "tsc",
+    srcs = ["tsc.bzl"],
+    deps = [
+        # this is a bzl_library target, exposing the package_json.bzl file we depend on
+        "@npm//:typescript",
+    ],
+)
+
+stardoc_with_diff_test(
+    name = "tsc-docs",
+    bzl_library_target = ":tsc",
+)
+
+update_docs(name = "docs")
+```
```

Comment:
Do we have a workable example in the `bazel_examples` repo? If we do have, we can add a link here.